recipe/meta.yaml

```diff
@@ -1,90 +1,57 @@
+  string: mps_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                                 # [gpu_variant == "metal"]
+  missing_dso_whitelist:                                                                         # [s390x or (gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
+    - "**/libcuda.so*"                                                                           # [(gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
+    - '$RPATH/ld64.so.1'  # [s390x]
+  
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}                        # [(cuda_compiler_version or "").startswith("12")]
+    - {{ compiler('cuda') }}                              # [gpu_variant == "cuda-12"]
     - cmake
-    - git
-    - ninja
+    - ninja-base
     - pkgconfig
+    - patch     # [unix]
+    - m2-patch  # [win]
   host:
-    - cudatoolkit      {{ cuda_compiler_version }}  # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
-    - cuda-version     {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
-    - cuda-cudart-dev  {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
-    - libcublas-dev    {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
-
-    - blas-devel * *{{ blas_impl }}                 # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
-    - mkl-devel {{ mkl }}                           # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
+    - cudatoolkit      {{ cuda_compiler_version }}        # [gpu_variant == "cuda-11"]
+    - cuda-version     {{ cuda_compiler_version }}        # [(gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
+    - cuda-cudart-dev  {{ cuda_compiler_version }}        # [gpu_variant == "cuda-12"]
+    - libcublas-dev    {{ cuda_compiler_version }}        # [gpu_variant == "cuda-12"]
+    - openblas-devel {{ openblas }}                       # [blas_impl == "openblas"]
+    - mkl-devel {{ mkl }}                                 # [blas_impl == "mkl"]
+    - pthread-stubs 0.3                                   # [linux]
   run:
-    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}  # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
-    - {{ pin_compatible('cuda-version', max_pin='x.x') }} # [cuda_compiler_version != "None"]
-
-    - llvm-openmp                                   # [linux and cuda_compiler_version == "None" and blas_impl == "mkl"]
-    - __cuda >={{ cuda_compiler_version }}          # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
+    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}  # [gpu_variant == "cuda-11"]
+    - {{ pin_compatible('cuda-version', max_pin='x.x') }} # [(gpu_variant == "cuda-11") or (gpu_variant == "cuda-12")]
 
 test:
   commands:
```

Comment:
Should there be test commands to verify files exist?
```
  - test -f $PREFIX/include/llama.h    # [unix]
  - test -f $PREFIX/bin/main               # [unix]
  - test -f $PREFIX/bin/server            # [unix]
  - test -f $PREFIX/lib/libllama.so     # [linux]
  - test -f $PREFIX/lib/libllama.dylib   # [osx]
  - if not exist %PREFIX%/Library/include/llama.h exit 1    # [win]
  - if not exist %PREFIX%/Library/lib/llama.lib exit 1    # [win]
  - if not exist %PREFIX%/Library/bin/llama.dll exit 1    # [win]
  - if not exist %PREFIX%/Library/bin/main.exe exit 1    # [win]
  - if not exist %PREFIX%/Library/bin/server.exe exit 1   # [win]
```

Reply:
Thanks, addressed